maven/plugins/com.liferay.ide.maven.core/src/com/liferay/ide/maven/core/NewMavenJSFModuleProjectProvider.java

```diff
@@ -0,0 +1,278 @@
+
+package com.liferay.ide.maven.core;
+
+import com.liferay.ide.core.util.CoreUtil;
+import com.liferay.ide.project.core.NewLiferayProjectProvider;
+import com.liferay.ide.project.core.modules.jsf.NewLiferayJSFModuleProjectOp;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.apache.http.util.EntityUtils;
+import org.apache.maven.archetype.catalog.Archetype;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.m2e.core.MavenPlugin;
+import org.eclipse.m2e.core.project.IProjectConfigurationManager;
+import org.eclipse.m2e.core.project.ProjectImportConfiguration;
+import org.eclipse.m2e.core.project.ResolverConfiguration;
+import org.eclipse.sapphire.platform.PathBridge;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.osgi.framework.Version;
+
+/**
+ * @author Simon Jiang
+ */
+
+@SuppressWarnings( "restriction" )
+public class NewMavenJSFModuleProjectProvider extends LiferayMavenProjectProvider
+    implements NewLiferayProjectProvider<NewLiferayJSFModuleProjectOp>
+{
+
+    private final static Map<String, String> JSF_TEMPLATE_MAP = new HashMap<String, String>();
+
+    public NewMavenJSFModuleProjectProvider()
+    {
+        super();
+        JSF_TEMPLATE_MAP.put( "JSF Standard", "jsf" );
+        JSF_TEMPLATE_MAP.put( "Liferay Faces Alloy", "alloy" );
+        JSF_TEMPLATE_MAP.put( "ICEFaces", "icefaces" );
+        JSF_TEMPLATE_MAP.put( "PrimeFaces", "primefaces" );
+        JSF_TEMPLATE_MAP.put( "RichFaces", "richfaces" );
+    }
+
+    @Override
+    public IStatus createNewProject( NewLiferayJSFModuleProjectOp op, IProgressMonitor monitor ) throws CoreException
+    {
+        IStatus retval = null;
+
+        final IProjectConfigurationManager projectConfigurationManager = MavenPlugin.getProjectConfigurationManager();
+
+        final String projectName = op.getProjectName().content();
+
+        IPath location = PathBridge.create( op.getLocation().content() );
+
+        // for location we should use the parent location
+        if( location.lastSegment().equals( projectName ) )
+        {
+            // use parent dir since maven archetype will generate new dir under this location
+            location = location.removeLastSegments( 1 );
+        }
+
+        final String groupId = op.getGroupId().content();
```

Comment:
All of this code is duplicated from above.  We shouldn't have nearly the same code twice.